azure-pipelines.yml

```diff
@@ -43,6 +43,40 @@ jobs:
         name: 'Downgrade pytest'
         cmd: 'pip install "pytest<5"'
 
+- job: micropython
+  pool:
+    vmImage: 'Ubuntu-16.04'
+  steps:
+    - script: sudo apt-get install -y build-essential curl libreadline-dev libffi-dev git pkg-config gcc-arm-none-eabi libnewlib-arm-none-eabi
+      displayName: Install micropython dependencies
+
+    - script: git clone --recurse-submodules https://github.com/micropython/micropython.git micropython
+      displayName: Clone micropython repo
+
+    - script: sh -c 'cd micropython/mpy-cross && make'
+      displayName: Make mpy-cross
+
+    - script: sh -c 'cd micropython/ports/unix && make axtls && make'
+      displayName: Make MicroPython
+
+    - script: micropython/ports/unix/micropython -c 'import upip; upip.install(["unittest"])'
+      displayName: Install tooling
+
+    - script: micropython/ports/unix/micropython -X heapsize=50M -m unittest
+      displayName: Run unittest
+
+#    - script: curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
```

Comment:
these are not needed?

Reply:
Only if we try to go back to pytest (unlikely).

Definitely can remove, will do so.

Reply:
Done.

I think the framework is in place now for MicroPython. Shall I go ahead and merge this PR, even though there's no actual `unittest` suite yet, so that it's in `master` for both of us to work from?

Reply:
Ya, good to have multiple heads working on it.